samples/EchoBot/EchoBot.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Middleware;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Samples.Echo
+{
+    public class EchoState : StoreItem
+    {
+        public string eTag { get; set; }
+
+        public int TurnNumber { get; set; }
+    }
+
+    public class EchoBot : IBot
+    {
+        private readonly IMyService _myService;
+
+        public EchoBot(IMyService myService)
+        {
+            _myService = myService;
+        }
+
+        public async Task OnReceiveActivity(IBotContext context)
+        {
+            var msgActivity = context.Request.AsMessageActivity();
+            if (msgActivity != null)
+            {
+                var conversationState = context.GetConversationState<EchoState>() ?? new EchoState();
```

Comment:
Does this work? It seems you're not saving Conversation State, in the event you create a new instance of EchoState

Reply:
should work

---
In reply to: [171711609](https://github.com/Microsoft/botbuilder-dotnet/pull/218#discussion_r171711609) [](ancestors = 171711609)

Reply:
Yep, the API saves changes to any state that was materialized.